src/Handler/CacheTrait.php

```diff
@@ -223,6 +223,10 @@ public function __invoke(RequestInterface $request, array $options)
             $ttl = (!empty($options['cache_ttl'])) ? $options['cache_ttl'] : null;
 
             $this->cacheResponse($request, $response, $ttl);
+
+            // we need to rewind the response body, because the response is a stream and the stream
+            // is already read in the cache response method
+            $response->getBody()->rewind();
```

Comment:
Should be done in `cacheResponse` IMHO, I expect this method to leave my object unchanged 🤔

Reply:
:+1: it makes more sense to me!